created `:active-view-transition` pseudo-class page (#40404)
mdn/content: files/en-us/web/css/_colon_active-view-transition/index.md

---
title: :active-view-transition
slug: Web/CSS/:active-view-transition
page-type: css-pseudo-class
browser-compat: css.selectors.active-view-transition
sidebar: cssref
---

The **`:active-view-transition`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches the root element of a document when a [view transition](/en-US/docs/Web/API/View_Transition_API#concepts_and_usage) is in progress (_active_) and stops matching once the transition has completed.

## Syntax

```css
:root:active-view-transition ... {
  /* ... */
}
```

## Examples

> ### Styling a view transition in progress

Reviewer: ```suggestion
### Styling an active view transition
```

This example extends on the [same-document view transition](/en-US/docs/Web/API/Document/startViewTransition#same-document_view_transition) example on the `startViewTransition` page.

```html
<main>
  <section class="color">
    <h2>Color is changing!</h2>
  </section>
  <button id="change-color">Change Color</button>
</main>
```

> A `<h2>` element initially has a `display: none` style, and this is overwritten using the `:active-view-transition` pseudo-class setting the `<h2>` style to `display: block`, and the button is also hidden using `visibility:hidden`, when the view transition is in progress:

Reviewer: ```suggestion
A `<h2>` element initially has a `display: none` style, and this is overwritten using the `:active-view-transition` pseudo-class setting the `<h2>` style to `display: block`.
The button is hidden using `visibility: hidden`, when the view transition is in progress:
```

---
title: :active-view-transition
slug: Web/CSS/:active-view-transition
page-type: css-pseudo-class
browser-compat: css.selectors.active-view-transition
sidebar: cssref
---

The **`:active-view-transition`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches the root element of a document when a [view transition](/en-US/docs/Web/API/View_Transition_API#concepts_and_usage) is in progress (_active_) and stops matching once the transition has completed.

## Syntax

```css
:root:active-view-transition ... {
  /* ... */
}
```

## Examples

### Styling a view transition in progress

This example extends on the [same-document view transition](/en-US/docs/Web/API/Document/startViewTransition#same-document_view_transition) example on the `startViewTransition` page.

#### HTML

```html
<main>
  <section class="color">
    <h2>Color is changing!</h2>
  </section>
  <button id="change-color">Change Color</button>
</main>
```

#### CSS

> The heading is set to `display: none` and this rule is overwritten when the view transition starts using the `:active-view-transition` pseudo-class:

Reviewer: ```suggestion
A `<h2>` element initially has a `display: none` style, and this is overwritten using the `:active-view-transition` pseudo-class when the view transition is in progress, setting the `<h2>` style to `display: block`:
```

---
title: :active-view-transition
slug: Web/CSS/:active-view-transition
page-type: css-pseudo-class
browser-compat: css.selectors.active-view-transition
sidebar: cssref
---

The **`:active-view-transition`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches elements when a [view transition](/en-US/docs/Web/API/View_Transition_API#concepts_and_usage) is in progress (_active_) and stops to match once the transition has completed.

> This pseudo-class is useful for styling elements while a transition is in progress.
> 

Reviewer: ```suggestion
```

## Syntax

```css
:root:active-view-transition ... {
  /* ... */
}
```

## Examples

### Showing View Transition is running

> This example extends on the [Color Change SPA View Transition](/en-US/docs/Web/API/Document/startViewTransition#color_change_spa_view_transition) on the `startViewTransition` page.

Reviewer: ```suggestion
This example extends on the [same-document view transition](/en-US/docs/Web/API/Document/startViewTransition#same-document_view_transition) on the `startViewTransition` page.
```
